src/rpc/blockchain.cpp

```diff
@@ -1501,34 +1501,140 @@ UniValue MempoolInfoToJSON(const CTxMemPool& pool)
     ret.pushKV("mempoolminfee", ValueFromAmount(std::max(pool.GetMinFee(maxmempool), ::minRelayTxFee).GetFeePerK()));
     ret.pushKV("minrelaytxfee", ValueFromAmount(::minRelayTxFee.GetFeePerK()));
     ret.pushKV("unbroadcastcount", uint64_t{pool.GetUnbroadcastTxs().size()});
+
+    if (feeLimits) {
+        const MempoolHistogramFeeRates& limits = feeLimits.value();
+
+        /* Keep histogram per...
+         * ... cumulated tx sizes
+         * ... txns (count)
+         * ... cumulated fees */
+        std::vector<uint64_t> sizes(limits.size(), 0);
+        std::vector<uint64_t> count(limits.size(), 0);
+        std::vector<CAmount> fees(limits.size(), 0);
+
+        for (const CTxMemPoolEntry& e : pool.mapTx) {
+            const CAmount fee{e.GetFee()};
+            const CAmount afees{e.GetModFeesWithAncestors()};
+            const CAmount dfees{e.GetModFeesWithDescendants()};
+
+            const int64_t size{static_cast<int64_t>(e.GetTxSize())};
+            const int64_t asize{static_cast<int64_t>(e.GetSizeWithAncestors())};
+            const int64_t dsize{static_cast<int64_t>(e.GetSizeWithDescendants())};
+
+            CHECK_NONFATAL(size > 0);
+            CHECK_NONFATAL(asize > 0);
+            CHECK_NONFATAL(dsize > 0);
+            CHECK_NONFATAL(asize + dsize - size > 0);
```

Comment:
Drive-by comment: now that you're constructing the fee rates properly with the `CFeeRate` ctor, there is no longer a need for these greater than zero checks. Will re-review the rest.

Reply:
Yes, good point. Thank you